x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -159,13 +159,18 @@ export const InnerWorkspacePanel = React.memo(function InnerWorkspacePanel({
     () => {
       if (!configurationValidationError?.length) {
         try {
-          return buildExpression({
+          const ast = buildExpression({
             visualization: activeVisualization,
             visualizationState,
             datasourceMap,
             datasourceStates,
             datasourceLayers: framePublicAPI.datasourceLayers,
           });
+          if (ast) {
+            return toExpression(ast);
```

Comment:
nit: I'd put a comment here explaining why we return a string. Otherwise one might want to refactor later without paying attention to fetches.